src/loaders/zarrLoader.js

```diff
@@ -60,18 +63,30 @@ export default class ZarrLoader {
    * @param {Array} loaderSelection, Array of valid dimension selections
    * @returns {Object} data: TypedArray[], width: number (tileSize), height: number (tileSize)
    */
-  async getTile({ x, y, z, loaderSelection = [] }) {
+  async getTile({ x, y, z, loaderSelection = [], signal }) {
+    const { TypedArray } = DTYPE_VALUES[this.dtype];
     const source = this._getSource(z);
     const [xIndex, yIndex] = ['x', 'y'].map(k => this._dimIndices.get(k));
 
     const dataRequests = loaderSelection.map(async sel => {
       const chunkKey = this._serializeSelection(sel);
       chunkKey[yIndex] = y;
       chunkKey[xIndex] = x;
-      const {
-        data,
-        shape: [height, width]
-      } = await source.getRawChunk(chunkKey);
+
+      const key = source.keyPrefix + chunkKey.join('.');
+      const buffer = await source.store.getItem(key, { signal });
+      if (source.store instanceof HTTPStore && signal?.aborted) return null;
```

Comment:
If `signal.aborted`, then `dataRequests` == `[null, null, null, null]`. Should `getTile` also return `null` in this case? Instead of `{ data, height, width }`?

Reply:
I think that's the pattern in the OME-TIFF Loader.

Reply:
I also discovered a bug in deck.gl so we're going to wait for that: https://github.com/visgl/deck.gl/issues/5041. I will fix this though.

Reply:
I looked into this as well @manzt.  I realized why I was confused about the `httpStore` not having error handling - deck.gl is handling that now.  See [here](https://deck.gl/docs/api-reference/geo-layers/tile-layer#gettiledata) where it says that the `TileLayer` handles the error:

>It is recommended to pass signal to any fetch calls and check its aborted property before doing any expensive computation. If signal is aborted, then throw or return falsy from getTileData so the data is not cached; do not return incomplete data. If signal is aborted, but getTileData still returns a truthy response, then its data will be cached.

So right now, we never actually hit this line because the `fetch` call throws an `AbortError`.  I returned null from `geotiff.js` (instead of allowing the error to propagate) because `geotiff.js` has a somewhat complicated fetching mechanism that doesn't seem to allow nice error propagation.  But it's fine to allow the error to propagate.  It's up to you if you want this to return null instead of that - we can add the error handling into the store or the loader, your call.  Does that make sense?

Reply:
The bug has been resolved now so I think this is the last thing outstanding @manzt.